### DISCUSSION TITLE
How to generate a macOS windowed app (BUNDLE) in spec file?

### DISCUSSION THREAD

**Discussion by joncox123** [pyinstaller/pyinstaller#8444] [Help] [answered]
I am trying to create macOS app (windowed app with info.plist, not a UNIX executable) using a spec file. I do not want to use the command line. The spec file documentation [describes creating a "BUNDLE" app](https://pyinstaller.org/en/stable/spec-files.html#spec-file-options-for-a-macos-bundle) from the "exe", but it does not say what to do with this. I tried passing this BUNDLE to "COLLECT", but it throws the following error:
`TypeError: Invalid argument type for COLLECT: <class 'PyInstaller.building.osx.BUNDLE'>`

What is the proper way to make use of the macOS BUNDLE? The documentation lacks the proper context.

```
# Do what with this in spec file??
app = BUNDLE(exe,
         name='myscript.app',
         icon=None,
         bundle_identifier=None)
```

**Comment by rokm** [comment, marked as answer]
Hmm, the part of docs you linked is actually outdated - you should not be generating BUNDLE from exe, unless you are trying to create a onefile .app bundle (which you should not).

Instead, you should pass instance of `COLLECT` (`coll`) to the BUNDLE:

```
...

coll = COLLECT(...)

app = BUNDLE(
    coll,
    name='myscript.app',
    icon=None,
    bundle_identifier=None,
)
```

The most surefire way to find this out would be to use command-line to generate a sample spec for you, even if you don't want to use command-line later on...

---

> but it does not say what to do with this

It will generate `myscript.app` in the `dist` directory, *in addition* to the POSIX executable.

**Reply by joncox123** [reply]
Thank you for clarifying!
